Extra inputs are not permitted if `fill ... --traces`

### Metadata

 * Python: 
```
user@mac execution-specs % .venv/bin/python3 --version
Python 3.12.12
```
 * Operating System: osx

#### `pip freeze`

No pip, stale template?

Fails in 0ebfc287

### What was wrong?

Fill a test, any test, with tracing on:

```bash
uv run fill --clean -vv -m blockchain_test 'tests/osaka/eip7883_modexp_gas_increase/test_modexp_thresholds.py::test_modexp_variable_gas_cost[fork_Osaka-blockchain_test_from_state_test-S0]' --fork=Osaka --traces
```

Fails with:

```bash
Executing: pytest -c /Users/user/sources/execution-specs/packages/testing/src/execution_testing/cli/pytest_commands/pytest_ini_files/pytest-fill.ini --rootdir . --clean -vv -m 
blockchain_test tests/osaka/eip7883_modexp_gas_increase/test_modexp_thresholds.py::test_modexp_variable_gas_cost --fork=Osaka --traces --ignore tests/evm_tools --ignore 
tests/json_infra --ignore tests/fixtures
============================================================================== test session starts ===============================================================================
platform darwin -- Python 3.12.12, pytest-8.4.2, pluggy-1.6.0 -- /Users/user/sources/execution-specs/.venv/bin/python
Generating fixtures for: Osaka, PragueToOsakaAtTime15k
2.19.0
cachedir: .pytest_cache
Log file: logs/fill-20260109-152234-main.log
metadata: {'Python': '3.12.12', 'Platform': 'macOS-15.6-arm64-arm-64bit', 'Packages': {'pytest': '8.4.2', 'pluggy': '1.6.0'}, 'Plugins': {'regex': '0.2.0', 'html': '4.1.1', 'xdist': '3.8.0', 'json-report': '1.5.0', 'metadata': '3.1.1', 'cov': '4.1.0', 'custom-report': '1.0.1'}, 'Tools': {'t8n': '2.19.0'}, 'Command-line args': '<code>fill -c /Users/user/sources/execution-specs/packages/testing/src/execution_testing/cli/pytest_commands/pytest_ini_files/pytest-fill.ini --rootdir . --clean -vv -m blockchain_test tests/osaka/eip7883_modexp_gas_increase/test_modexp_thresholds.py::test_modexp_variable_gas_cost[fork_Osaka-blockchain_test_from_state_test-S0] --fork=Osaka --traces --ignore tests/evm_tools --ignore tests/json_infra --ignore tests/fixtures</code>'}
rootdir: /Users/user/sources/execution-specs
configfile: packages/testing/src/execution_testing/cli/pytest_commands/pytest_ini_files/pytest-fill.ini
plugins: regex-0.2.0, html-4.1.1, xdist-3.8.0, json-report-1.5.0, metadata-3.1.1, cov-4.1.0, custom-report-1.0.1
collecting ...  pytest-regex selected 1 tests to run for regex: .*
collected 1 item                                                                                                                                                                 

tests/osaka/eip7883_modexp_gas_increase/test_modexp_thresholds.py::test_modexp_variable_gas_cost[fork_Osaka-blockchain_test_from_state_test-S0] FAILED                      [1/1]

==================================================================================== FAILURES ====================================================================================
__________________________________________________ test_modexp_variable_gas_cost[fork_Osaka-blockchain_test_from_state_test-S0] __________________________________________________
tests/osaka/eip7883_modexp_gas_increase/test_modexp_thresholds.py:700: in test_modexp_variable_gas_cost
    state_test(pre=pre, tx=tx, post=post)
packages/testing/src/execution_testing/cli/pytest_commands/plugins/filler/filler.py:1385: in __init__
    fixture = self.generate(
packages/testing/src/execution_testing/specs/state.py:503: in generate
    return self.generate_blockchain_test().generate(
packages/testing/src/execution_testing/specs/blockchain.py:1010: in generate
    return self.make_fixture(t8n)
           ^^^^^^^^^^^^^^^^^^^^^^
packages/testing/src/execution_testing/specs/blockchain.py:805: in make_fixture
    built_block = self.generate_block_data(
packages/testing/src/execution_testing/specs/blockchain.py:600: in generate_block_data
    transition_tool_output = t8n.evaluate(
packages/testing/src/execution_testing/client_clis/clis/execution_specs.py:152: in evaluate
    self.collect_traces(
packages/testing/src/execution_testing/client_clis/transition_tool.py:223: in collect_traces
    traces.append(TransactionTraces.from_file(trace_file_path))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
packages/testing/src/execution_testing/client_clis/cli_types.py:111: in from_file
    TraceLine.model_validate_json(line) for line in trace_lines
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   pydantic_core._pydantic_core.ValidationError: 1 validation error for TraceLine
E   memory
E     Extra inputs are not permitted [type=extra_forbidden, input_value='0x0000000000000000000000...00000000000000000000000', input_type=str]
E       For further information visit https://errors.pydantic.dev/2.12/v/extra_forbidden
================================================================================ warnings summary ================================================================================
.venv/lib/python3.12/site-packages/_pytest/config/__init__.py:812
  /Users/user/sources/execution-specs/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py:812: PytestAssertRewriteWarning: Module already imported so cannot be rewritten; execution_testing.cli.pytest_commands.plugins.shared.execute_fill
    self.import_plugin(arg, consider_entry_points=True)

.venv/lib/python3.12/site-packages/_pytest/config/__init__.py:812
  /Users/user/sources/execution-specs/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py:812: PytestAssertRewriteWarning: Module already imported so cannot be rewritten; execution_testing.cli.pytest_commands.plugins.forks.forks
    self.import_plugin(arg, consider_entry_points=True)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
------------------------------------------------ Log file: /Users/user/sources/execution-specs/logs/fill-20260109-152234-main.log ------------------------------------------------
--------------------------------------- Generated html report: file:///Users/user/sources/execution-specs/fixtures/.meta/report_fill.html ----------------------------------------
============================================================================ short test summary info =============================================================================
FAILED tests/osaka/eip7883_modexp_gas_increase/test_modexp_thresholds.py::test_modexp_variable_gas_cost[fork_Osaka-blockchain_test_from_state_test-S0] - pydantic_core._pydantic_core.ValidationError: 1 validation error for TraceLine
memory
  Extra inputs are not permitted [type=extra_forbidden, input_value='0x0000000000000000000000...00000000000000000000000', input_type=str]
    For further information visit https://errors.pydantic.dev/2.12/v/extra_forbidden
========================================================================= 1 failed, 2 warnings in 1.30s ==========================================================================
```

Filling done with EELS.

### How can it be fixed?

Without `--traces` test fills correctly. This fails in 0ebfc287 while it works in 9e56ac97.

--- a/packages/testing/src/execution_testing/client_clis/cli_types.py
+++ b/packages/testing/src/execution_testing/client_clis/cli_types.py
@@ -67,6 +67,8 @@ class RejectedTransaction(CamelModel):
 class TraceLine(CamelModel):
     """Single trace line contained in the traces output."""
 
+    model_config = CamelModel.model_config | {"extra": "ignore"}
+
     pc: int
     op: int
     gas: HexNumber

--- a/packages/testing/src/execution_testing/rpc/rpc_types.py
+++ b/packages/testing/src/execution_testing/rpc/rpc_types.py
@@ -64,6 +64,7 @@ class TransactionByHashResponse(Transaction):
 
     gas_limit: HexNumber = Field(HexNumber(21_000), alias="gas")
     transaction_hash: Hash = Field(..., alias="hash")
+    transaction_index: HexNumber | None = None
     sender: EOA | None = Field(None, alias="from")
 
     # The to field can have different names in different clients, so we use

--- a/packages/testing/src/execution_testing/rpc/rpc_types.py
+++ b/packages/testing/src/execution_testing/rpc/rpc_types.py
@@ -57,6 +57,8 @@ class JSONRPCError(Exception):
 class TransactionByHashResponse(Transaction):
     """Represents the response of a transaction by hash request."""
 
+    model_config = Transaction.model_config | {"extra": "ignore"}
+
     block_hash: Hash | None = None
     block_number: HexNumber | None = None
 

--- a/packages/testing/src/execution_testing/test_types/transaction_types.py
+++ b/packages/testing/src/execution_testing/test_types/transaction_types.py
@@ -305,7 +305,10 @@ class Transaction(
     def strip_hash_from_t8n_output(cls, data: Any) -> Any:
         """Strip the hash field which may be included in t8n tool output."""
         if isinstance(data, dict):
-            data.pop("hash", None)
+            # If the class has a transaction_hash, keep it. This is likely
+            # an internal RPC class that extends from Transaction.
+            if "transaction_hash" not in cls.model_fields:
+                data.pop("hash", None)
         return data
 
     gas_limit: HexNumber = Field(

